src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -172,6 +206,70 @@ public static Text of(Object... objects) {
     }
 
     /**
+     * Creates a new Text instance with all {@link Placeholder}s replaced. All
+     * placeholders without a non-null replacement are ignored. All replacements
+     * will be wrapped in a {@link Text} using {@link Texts#of(Object...)} the
+     * color and the style from the placeholder are transfered to that method as
+     * well.
+     *
+     * @param template The template text in which all {@link Placeholder}s
+     *        should be replaced
+     * @param replacements The values available to replace the placeholders
+     * @return The text with all possible placeholders replaced
+     */
+    public static Text format(Text template, Map<String, ?> replacements) {
+        checkNotNull(template, "template");
+        checkNotNull(replacements, "values");
+        if (replacements.isEmpty()) {
+            return template;
+        }
+        return formatNoChecks(template, replacements);
+    }
+
+    /**
+     * Creates a new Text instance with all {@link Placeholder}s replaced. All
+     * placeholders without a non-null replacement are ignored. All replacements
+     * will be wrapped in a {@link Text} using {@link Texts#of(Object...)} the
+     * color and the style from the placeholder are transfered to that method as
+     * well.
+     *
+     * @param template The template text in which all {@link Placeholder}s
+     *        should be replaced
+     * @param replacements The values available to replace the placeholders. May
+     *        contain null values to skip the placeholder
+     * @return The text with all possible placeholders replaced
+     */
+    public static Text format(Text template, Object... replacements) {
+        checkNotNull(template, "template");
+        checkNotNull(replacements, "values");
+        Map<String, Object> replacementsMap = new HashMap<String, Object>();
+        int index = 0;
+        for (Object replacement : replacements) {
+            replacementsMap.put(Integer.toString(index++), replacement);
+        }
+        return formatNoChecks(template, replacementsMap);
+    }
+
+    private static Text formatNoChecks(Text template, Map<String, ?> replacements) {
+        // Is this a placeholder that should be replaced?
+        if (template instanceof Placeholder) {
+            Object replacement = replacements.get(((Placeholder) template).getKey());
+            // Only replace
+            if (replacement != null) {
+                // Copy color and style from placeholder
+                return Texts.of(template.getColor(), template.getStyle(), replacement);
```

Comment:
>>>A clickaction like OpenLink does not make sense with a variable text. And a hover text usually shows a hint/description for the object it describes.

>> What if a plugin adds something like an imgur link (I know of some servers that do) or some other variable link?

>The text with the open url action is usually provided as the param and not part of the template.
And if its part of the template then you will usually only replace a sub text.
If you need more explanation i can write an example code, but i cannot do that right now.

I get it now. Thanks for clarifying.